apps/site/assets/ts/models/__tests__/alert-test.ts

```diff
@@ -247,3 +249,56 @@ describe("hasAnActiveDiversion", () => {
     expect(hasAnActiveDiversion(stopId, [nonActiveDiversionAlert])).toBeFalsy();
   });
 });
+
+describe("alertsByRoute", () => {
+  test("it should group alerts across multiple routes", () => {
+    const alerts = [
+      {
+        id: "1234",
+        informed_entity: {
+          route: ["441442", "442"]
+        }
+      },
+      {
+        id: "4321",
+        informed_entity: {
+          route: ["442", "53"]
+        }
+      }
+    ] as Alert[];
+    const result = alertsByRoute(alerts);
+    expect(Object.keys(result).length).toBe(3);
+    expect(result["442"].length).toBe(2);
+    expect(result["441442"].length).toBe(1);
+    expect(result["53"].length).toBe(1);
+  });
+});
+
+describe("alertsByDirectionId", () => {
+  test("it should group alerts by directionId", () => {
+    const alerts = [
+      {
+        id: "1234",
+        informed_entity: {
+          direction_id: [0]
+        }
+      },
+      {
+        id: "4321",
+        informed_entity: {
+          direction_id: [null]
```

Comment:
question (non-blocking): It seems this test is additionally testing that alerts without direction_id are being filtered out. Is that intended? If so, may be worth mentioning in the test name?